src/spartan_router.erl

```diff
@@ -13,30 +13,26 @@
 -spec(upstreams_from_questions(dns:questions()) -> ordsets:ordset(upstream())).
 upstreams_from_questions([#dns_query{name=Name}]) ->
     Labels = spartan_app:parse_upstream_name(Name),
-    lists:map(fun normalize_ip/1, find_upstream(Name, Labels));
+    Upstreams = find_upstream(Name, Labels),
+    lists:map(fun validate_upstream/1, Upstreams);
 
 %% There is more than one question. This is beyond our capabilities at the moment
 upstreams_from_questions([Question|Others]) ->
     spartan_metrics:update([spartan, ignored_questions], length(Others), ?COUNTER),
     upstreams_from_questions([Question]).
 
-%% @private
-normalize_ip({NS, Port}) when is_list(NS) ->
-    {ok, IP} = inet:parse_ipv4_address(NS),
-    {IP, Port};
-normalize_ip({IP, Port}) when is_tuple(IP) andalso is_integer(Port) ->
```

Comment:
One potential flaw I see with removing this is that previously we were accepting any tuple as an ip address. Now we are forcing ipv4 tuple format as the only valid way. inet:ip6() tuple format is:

```
{0..65535,
     0..65535,
     0..65535,
     0..65535,
     0..65535,
     0..65535,
     0..65535,
     0..65535}
```

I also think that this can be handled in `validate_upstream` when we cross that bridge though, so I'm fine with this change.